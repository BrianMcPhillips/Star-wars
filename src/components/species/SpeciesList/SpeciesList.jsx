import React from 'react';
import PropTypes from 'prop-types';
import Species from '../Species/Species';

const SpeciesList = ({ data }) => {
  const listStuff = data.map((specie, i) =>
    <li key={specie.url + i}>
      <Species {...specie}/>
    </li> 
  );
  
  return (
    <ul>
      {listStuff}
    </ul>
  );
};

SpeciesList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default SpeciesList;
