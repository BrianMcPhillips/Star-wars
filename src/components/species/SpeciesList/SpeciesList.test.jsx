import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SpeciesList from './SpeciesList';

describe('SpeciesList component', () => {
  afterEach(() => cleanup());
  it('renders SpeciesList', () => {
    const { asFragment } = render(
      <SpeciesList data={[
        {
          name: 'Human',
          url: 'https://swapi.dev/api/species/1/'
        },
        {
          name: 'Human 2',
          url: 'https://swapi.dev/api/species/12/'
        }
      ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
