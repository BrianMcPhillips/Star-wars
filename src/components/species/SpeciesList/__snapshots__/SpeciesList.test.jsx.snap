// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeciesList component renders SpeciesList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1>
        Human
      </h1>
    </li>
    <li>
      <h1>
        Human 2
      </h1>
    </li>
  </ul>
</DocumentFragment>
`;
